specs/_features/eip7594/fork-choice.md

```diff
@@ -0,0 +1,122 @@
+# EIP-7594 -- Fork Choice
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Helpers](#helpers)
+  - [Modified `is_data_available`](#modified-is_data_available)
+  - [Modified `get_head`](#modified-get_head)
+- [Updated fork-choice handlers](#updated-fork-choice-handlers)
+  - [Modified `on_block`](#modified-on_block)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice accompanying EIP-7594.
+
+## Helpers
+
+### Modified `is_data_available`
+
+```python
+def is_data_available(beacon_block_root: Root) -> bool:
+    # `retrieve_column_sidecars` is implementation and context dependent, replacing
+    # `retrieve_blobs_and_proofs`. For the given block root, it returns all column 
+    # sidecars to sample, or raises an exception if they are not available. 
+    # The p2p network does not guarantee sidecar retrieval outside of 
+    # `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS` epochs.  
+    column_sidecars = retrieve_column_sidecars(beacon_block_root)
+    return all(
+        verify_data_column_sidecar_kzg_proofs(column_sidecar)
+        for column_sidecar in column_sidecars
+    )
+```
+
+### Modified `get_head`
+
+*Note*: children of the current `head` are required to be available in order to be considered by the fork-choice.
+
+```python
+def get_head(store: Store) -> Root:
+    # Get filtered block tree that only includes viable branches
+    blocks = get_filtered_block_tree(store)
+    # Execute the LMD-GHOST fork choice
+    head = store.justified_checkpoint.root
+    while True:
+        # Get available children for the current slot
+        children = [
+            root for (root, block) in blocks.items()
+            if (
+                block.parent_root == head
+                and is_data_available(root)
+            )
```

Comment:
The parentheses here aren't necessary, but I'm fine with keeping them for when we add back sampling if you want.

From your other PR:

<img width="623" alt="image" src="https://github.com/user-attachments/assets/e06a9927-5ce0-43cd-812b-43f5d3748a72">


```suggestion
            if block.parent_root == head and is_data_available(root):
```